ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -133,15 +152,16 @@ public IonParser(IonReader r, IOContext ctxt)
      */
     @Deprecated
     public IonParser(IonReader r, IOContext ctxt, ObjectCodec codec) {
-        this(r, IonSystemBuilder.standard().build(), ctxt, codec);
+        this(r, IonSystemBuilder.standard().build(), ctxt, codec, IonFactory.DEFAULT_ION_PARSER_FEATURE_FLAGS);
     }
 
-    IonParser(IonReader r, IonSystem system, IOContext ctxt, ObjectCodec codec) {
```

Comment:
We probably should retain this intermediate constructor (at least for 2.12.x); will add a deprecated variant after merge.